doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/index.md

```diff
@@ -2,304 +2,186 @@
+4. Perform build:
 
-1. Create a folder in $KAA\_HOME/listfiles/platform/ and name it using \[a-zA-z\_-\] symbols. You will be able to use the folder name as a value for the KAA\_PLATFORM parameter.
-2. Put the CMakeLists.txt file into the created folder. This file may contain specific compilation/linking flags, platform-dependent source files, third-party library dependencies, that is all information necessary for building the C endpoint SDK for this platform.
-3. Optionally, specify the following parameters in the CMakeLists.txt file.
-  * KAA\_INCLUDE\_PATHS - full path(s) to folder(s) containing additional header files
-  * KAA\_SOURCE\_FILES - full path(s) to additional source files
-  * KAA\_THIRDPARTY\_LIBRARIES - third-party libraries (the name of the library, for example, ssl, crypto)
+        make
 
-The following example illustrates the build procedure for the debug build with the INFO log level and disabled EVENTS feature.
+## Building Kaa application
 
-~~~
-mkdir build
-cd build
-cmake -DKAA_DEBUG_ENABLED=1 -DKAA_MAX_LOG_LEVEL=4 -DKAA_WITHOUT_EVENTS=1 ..
-make
-make install
-~~~
+This section is about how to build your Kaa aplication using C SDK.
 
-# C++ endpoint SDK
+In this guide we use *CMake* as a build system for the application, so focus on the writing the build system rather than the application's code. If you want master writing applciation code go to the Programming guide section. Before going on, make shure that all [dependencies](#dependencies) are installed.
 
-## Install build utils and dependencies
+1. First of all let's create some directories. The directories determine build structure. To keep it clean and transparent create three directories: my-kaa-application - a root directory, which contains all files, my-kaa-application/kaa - for the C SDK source files, my-kaa-application/src - for the application code.
 
-Before building the C++ endpoint SDK, install the following components on your machine:
+        mkdir my-kaa-application
+        cd my-kaa-application
+        mkdir kaa
+        mkdir src
 
-1. Install compilers:
-    1. For automatic installation, execute the following commands:
+2. Generate C SDK and unpack it to my-kaa-application/kaa directory:
 
-       ~~~ shell
-       $ sudo apt-get install g++
-       ~~~
+        tar -xvf c-sdk-archive.tar.gz -C my-kaa-application/kaa
 
-    2. For manual installation of version 4.8, refer to the following example:
+3. Create the application code. Create kaa-application.c file in my-kaa-application/src directory:
 
-       ~~~ shell
-       $ sudo apt-get install python-software-properties
-       $ sudo add-apt-repository ppa:ubuntu-toolchain-r/test
-       $ sudo apt-get update
-       $ sudo apt-get install g++-4.8
-       $ sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.8 50
-       ~~~
+        touch src/kaa-application.c
```

Comment:
Specify something like `vim` or `nano`. That gives a hint that user must write some code too.